packages/eslint-plugin-choreography-ts/src/no-unused-colocally-location.ts

```diff
@@ -0,0 +1,166 @@
+// no-unused-colocally-location rule
+// All locations specified for a `colocally` call must be used inside the body
+// https://github.com/shumbo/choreography-ts/issues/7
+
+"use strict";
+import { AST_NODE_TYPES, TSESLint, TSESTree } from "@typescript-eslint/utils";
+
+type MessageIDs = "warning";
+
+const choreographySelector = `VariableDeclaration[kind = "const"] > VariableDeclarator[id.typeAnnotation.typeAnnotation.typeName.name = "Choreography"]`;
+const colocallySelector = `${choreographySelector} CallExpression[callee.name = "colocally"]`;
+const colocallyColocallyUseLocater = `${colocallySelector} CallExpression[callee.name = "colocally"]`;
+const colocallyLocallyUseLocater = `${colocallySelector} CallExpression[callee.name = "locally"]`;
+const colocallyMulticastUseLocater = `${colocallySelector} CallExpression[callee.name = "multicast"]`;
+const colocallyCommUseLocater = `${colocallySelector} CallExpression[callee.name = "comm"]`;
+const colocallyBroadcastUseLocater = `${colocallySelector} CallExpression[callee.name = "broadcast"]`;
+const colocallyTypeUseLocater = `${colocallySelector} TSLiteralType`;
+const choreographyAtExit = `Program > ${choreographySelector}:exit`;
+
+// Implementation: Check whether each location passed to `colocally` is present inside the body of the choreography argument
+// as an argument to any operator that accepts location parameters (locally, colocally, comm, multicast, broadcast)
+// or as a type argument
+const noUnusedColocallyLocation: TSESLint.RuleModule<MessageIDs, []> = {
+  defaultOptions: [],
+  meta: {
+    type: "problem",
+    docs: {
+      description:
+        "All locations specified for `colocally` must be used inside the body of the call",
+      recommended: "recommended",
+      url: "https://github.com/shumbo/choreography-ts/issues/7",
+    },
+    fixable: undefined, // not automatically fixable
+    hasSuggestions: false, // no suggested fixes available
+    messages: {
+      warning: "Location `{{ location }}` is not used inside the body.",
+    },
+    schema: [],
+  },
+  create(context) {
+    // Store locations for each `colocally` usage
+    let Locations: {
```

Comment:
nits: probably use lowercase for normal variables
```suggestion
    let locations: {
```

Reply:
My mistake. It's fixed now.